README.md

```diff
@@ -109,14 +120,57 @@ where data in form of snapshots can be exported to, and imported from.
 ```sql
 -- A policy using the SNAPSHOT strategy.
 INSERT INTO "ext"."retention_policy"
-  (table_schema, table_name, partition_column, retention_period, target_repository_name, strategy)
+  (strategy, table_schema, table_name, partition_column, retention_period, target_repository_name)
 VALUES
-  ('doc', 'sensor_readings', 'time_month', 365, 'export_cold', 'snapshot');
+  ('snapshot', 'doc', 'sensor_readings', 'time_month', 365, 'export_cold');
 ```
 
 [implementation](cratedb_retention/strategy/snapshot.py) | [tutorial](https://community.crate.io/t/building-a-data-retention-policy-for-cratedb-with-apache-airflow/1001)
 
 
+## Internals
+
+The database schema for the retention policy table is defined like this.
+Each record represents a rule how to expire and retain data per table.
+At runtime, each record will correspond to a dedicated retention job task.
+
+```sql
+-- Set up the retention policy database table schema.
+CREATE TABLE IF NOT EXISTS {policy_table.fullname} (
+
+    -- Source: The database table operated upon.
+    "table_schema" TEXT,                        -- The source table schema.
+    "table_name" TEXT,                          -- The source table name.
+    "partition_column" TEXT NOT NULL,           -- The source table column name used for partitioning.
+
+    -- Retention parameters.
+    "retention_period" INTEGER NOT NULL,        -- Retention period in days.
+                                                -- TODO: Provide better description what this means.
+
+    -- Target: Where data is moved/relocated to.
+    "reallocation_attribute_name" TEXT,         -- TODO: Describe the meaning of this column value.
+    "reallocation_attribute_value" TEXT,        -- TODO: Describe the meaning of this column value.
+    "target_repository_name" TEXT,              -- The name of a repository created with `CREATE REPOSITORY ...`.
+
+    -- Strategy to apply for data retention.
+    "strategy" TEXT NOT NULL,
+
+    PRIMARY KEY ("table_schema", "table_name", "strategy")
+)
+CLUSTERED INTO 1 SHARDS;
+```
+
+You can add a retention policy by using an SQL statement like this.
+TODO: Describe what it does.
+```sql
+-- Add a retention policy using the DELETE strategy.
+INSERT INTO "ext"."retention_policy"
+  (strategy, table_schema, table_name, partition_column, retention_period)
+VALUES
+  ('delete', 'doc', 'raw_metrics', 'ts_day', 1);
+```
```

Comment:
See also GH-17, where I attempted to write down a REALLOCATE retention policy in human language.

Reply:
This retention policy will delete all data from `doc.raw_metrics` that is older than one day.

Reply:
Thank you. How exactly does this correlate with the `cutoff_day` argument obtained from the caller? Regularly, when called by the scheduler, you told me it would be "today". So, what would happen if that date would be in the past, or in the future?

Reply:
I would expect that:

- When `cutoff_day` is in the future, it does not have any effect, because, well, the cut-off date does not cut off any past data.
- When `cutoff_day` is in the past, the data considered to be expired will be limited up until that date. Newer data is not considered at all.

Are my assumptions correct?

Reply:
Right. So more precisely, it is:

> This retention policy will delete all data from doc.raw_metrics that is older than one day at the given cut-off date.

Reply:
Excellent. I've adjusted the documentation accordingly.